library/core/src/main/java/com/google/android/exoplayer2/audio/AudioCapabilities.java

```diff
@@ -158,4 +164,46 @@ public String toString() {
   private static boolean deviceMaySetExternalSurroundSoundGlobalSetting() {
     return Util.SDK_INT >= 17 && "Amazon".equals(Util.MANUFACTURER);
   }
+
+  @TargetApi(29)
+  private static int[] getDirectPlaybackSupportedEncodings() {
+      int[] possibleEncodings = new int[]{
```

Comment:
For now, could we restrict this to the useful subset that's compatible with `DefaultAudioSink`, which consists of the encodings AC3, E_AC3, E_AC3_JOC, AC4, DOLBY_TRUEHD, DTS, DTS_HD and PCM_16BIT?

Later on we can investigate making this class not immutable and having `supportsEncoding` call through directly to the platform so we don't need to worry about listing them exhaustively but that is beyond the scope of this change.

Reply:
OK, this sounds good.